### PR TITLE
Added solution.

### PR DESCRIPTION
**Pull request by Merkald**

### CODE REVIEW

File: src/main/java/core/basesyntax/Rectangle.java [hunk 1]
```diff
@@ -16,6 +16,32 @@ public Rectangle(Integer width, Integer length, String color) {
...
 
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+        if (obj == null) {
+            return false;
+        }
+        if (obj.getClass().equals(Rectangle.class)) {
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/Rectangle.java]:
```suggestion
        if (obj.getClass().equals(getClass())) {
```

File: src/main/java/core/basesyntax/Rectangle.java [hunk 2]
```diff
@@ -16,6 +16,32 @@ public Rectangle(Integer width, Integer length, String color) {
...
+        }
+        return false;
+    }
+
+    @Override
+    public int hashCode() {
+        int result = 13;
+        result = 31 * result + width.hashCode();
+        result = 31 * result + length.hashCode();
+        result = 31 * result + color.hashCode();
```
Comment (Tarabarnik) [inline src/main/java/core/basesyntax/Rectangle.java]: що буде, якщо якесь із полів == null?

Comment (Merkald) [reply src/main/java/core/basesyntax/Rectangle.java]: NullPointerException зараз переправлю

File: src/main/java/core/basesyntax/Rectangle.java [hunk 3]
```diff
@@ -16,6 +16,38 @@ public Rectangle(Integer width, Integer length, String color) {
...
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+        if (obj == null) {
+            return false;
+        }
+        if (obj.getClass().equals(Rectangle.class)) {
+            Rectangle rectangle = (Rectangle) obj;
+            return width.equals(((Rectangle) obj).width)
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/Rectangle.java]: do we need to cast every field of obj, if we have already rectangle object, can we use it here?

File: src/main/java/core/basesyntax/Rectangle.java (hunk 1 above)
Comment (RomanDubovskyi) [reply src/main/java/core/basesyntax/Rectangle.java]: have you fixed it ?

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request closed without merging**